app/models/dictionary.rb

```diff
@@ -499,10 +499,22 @@ def update_tags(tag_list)
       self.tags.create!(value: tag)
     end
 
-    tags_to_remove.each do |tag|
-      tag = self.tags.find_by(value: tag)
-      self.tags.delete(tag) if tag
+    tags_in_use = []
+    tags_to_remove.each do |tag_value|
+      tag = self.tags.find_by(value: tag_value)
+      if tag && tag.used_in_entries?
+        tags_in_use << tag_value
+      else
+        self.tags.delete(tag) if tag
+      end
```

Comment:
tagの存在チェックを2回しています。
次のように先頭にまとめると読みやすくなると思います。

```suggestion
      next unless tag
      
      if tag.used_in_entries?
        tags_in_use << tag_value
      else
        self.tags.delete(tag)
      end
```

ですが、そもそも`self.tags.find_by(value: tag_value)`で見つからないことはあり得るのでしょうか？

Reply:
ありがとうございます。
`tags_to_remove`が空とかnilだったら…と思って書きましたが、その場合はそもそもeachに入らないので必要なかったです。
なので、[重複チェック部分削除](https://github.com/pubannotation/pubdictionaries/pull/71/commits/efc696e834b92386ca524dd7f9baae63d00eff5d)にてご提案の修正から`next unless tag`を削除した形で修正しました。